data-raw/R2R_TMH_habitat_inputs/documentation/TMH_model_implementation_overview.Rmd

```diff
@@ -0,0 +1,260 @@
+  
+  annual_stats %>%
+    mutate(dist = round(cume_dist(-stat_in_duration), 3)) %>%
+    arrange(dist)
+  
+}
+
+calsim_30_day <- function(data, exceedance_function) {
+  dur_30_min_max <- exceedance_probs_monthly(data, roll_stat = roll_stat, annual_stat = annual_stat)
+  
+  interpolate_probs_30_min_max <- approxfun(x = dur_30_min_max$dist, 
+                                            dur_30_min_max$stat_in_duration)
+  
+  d30 <- interpolate_probs_30_min_max(0.5) 
+  
+  return(d30)
+}
+
+```
+
+```{r include=FALSE}
+watershed_input <- "Cosumnes River"
+
+flood_flow_ex = DSMflow::flows_cfs$biop_itp_2018_2019 |> 
+      filter(date >= as_date("1980-01-01")) |> 
+      select(watershed_input, date) |> 
+      rename(flow_cfs = watershed_input) 
+    
+exceedance_probs_monthly_fun <- exceedance_probs_monthly(flood_flow_ex, 'min', 'max')
+calsim_30_day(flood_flow_ex, exceedance_probs_monthly_fun)
+
+ggplot(exceedance_probs_monthly_fun, aes(x = dist, y = stat_in_duration)) + 
+  geom_line() + 
+  ggtitle('interpolation example: want 50% distance value')
+    
+```
+
+```{r echo=FALSE}
+# plot it: 
+ggplot(DSMflow::flows_cfs$biop_itp_2018_2019 |> 
+      filter(date >= as_date("1980-01-01")) |> 
+      select(watershed_input, date) |> 
+      rename(flow_cfs = watershed_input)) +
+  geom_line(aes(x = date, y = flow_cfs)) +
+  geom_hline(aes(yintercept = calsim_30_day(flood_flow_ex, exceedance_probs_monthly_fun), color = "2 year flow"), linetype = "dashed") + 
+  ggtitle('Example of 2 Year Flow on the Consumnes River', subtitle = "used to scale floodplain habitat") +
+  theme_minimal() + 
+  theme(legend.position="top", 
+        legend.title = element_blank())
+
+```
+
+**2) Calculate the proportion increase of habitat at flow point of reference**
+
+The proportion increase between baseline and theoretical maximum habitat was calculated using the following equation. The `existing_acres` were calculated using the median (for in-channel) and 2-year flows (for floodplain).
+
+`proportion_increase = (max_hab_acres - existing_acres) / existing_acres + 1`
+
+**3) Scale existing habitat by proportion increase**
+
+The existing acres are then scaled for each watershed (`ws`) by the calculated `proportion_increase`.
+
+`new_hab_acres <- DSMhabitat::fr_fp$biop_itp_2018_2019[ws , , ] * proportion_increase`
+
+**4) Save new data objects for Theoretical Maximum Habitat**
+
+The new data objects are saved in the package as:
+
+-   `DSMhabitat::fr_spawn$r_to_r_tmh`
+-   `DSMhabitat::fr_juv$r_to_r_tmh`
+-   `DSMhabitat::fr_fry$r_to_r_tmh`
+-   `DSMhabitat::fr_fp$r_to_r_tmh`
+-   `DSMhabitat::fr_delta$r_to_r_tmh` *See [TODO link]* for details on Delta methodology for theoretical maximum habitat
+
+#### Comparison plots between Baseline and Theoretical Maximum Habitat
+
+##### Spawning Habitat: 
+
+```{r echo=FALSE}
+
+## spawning plot:  ---------------------------------------------------------
+r_to_r_max_habitat <- DSMhabitat::fr_spawn$r_to_r_tmh |> 
```

Comment:
Labels could be imrpved here "Acres of Spawning Habitat" or add a title/

Reply:
please check title/labels for all the plots

Reply:
done